### DISCUSSION TITLE
button style question

### DISCUSSION THREAD

**Discussion by BonanzaBegin** [AvaloniaUI/Avalonia#10916] [Q&A]
When my command can not be executed, the button style is not in the disabled style state, but simply cannot be clicked. How do I set this to make the button style in the Disabled state when CanExecute=false for the command?

**Comment by BonanzaBegin** [comment]
![c1d5c3cd3de1d6b19f7fc85ba4cfd7f](https://user-images.githubusercontent.com/58451874/230055530-4f2a3743-72c7-45b6-832c-7ca740455b15.png)
![dfd0ba899b6e97416f687ef56889fa9](https://user-images.githubusercontent.com/58451874/230055562-66e8e154-150a-4701-bb71-20d6ee6174b1.png)
here are the images

**Comment by BonanzaBegin** [comment]
![c665fd1702c868ebd5b2e2c55b25a86](https://user-images.githubusercontent.com/58451874/230055825-a8bccbd7-2957-43f1-ad48-dd74df0f48a1.png)

**Reply by timunie** [reply]
Hm normally that should work. Can you show how your command is defined? Also please enable compiled bindings to debug binding issues.

**Comment by maxkatz6** [comment]
When command cannot be executed, button has :disabled pseudoclass, which also applies custom styling for both simple and fluent themes.

**Comment by dbriard** [comment]
May be just forgot to call NotifyCanExecuteChanged() on the command to refresh the state of the button?
